File: backend/src/models/organization.rs
use serde::{Deserialize, Serialize};
use time::OffsetDateTime;
use uuid::Uuid;
use validator::Validate;

#[derive(Debug, Clone, Serialize, sqlx::FromRow)]
pub struct Organization {
    pub id: Uuid,
    pub name: String,
    pub slug: String,
    pub timezone: String,
    #[serde(with = "time::serde::rfc3339")]
    pub created_at: OffsetDateTime,
    #[serde(with = "time::serde::rfc3339")]
    pub updated_at: OffsetDateTime,
}

#[derive(Debug, Deserialize, Validate)]
pub struct UpdateOrganizationRequest {
    #[validate(length(max = 100))]
    pub name: Option<String>,
    #[validate(length(max = 100))]
    pub timezone: Option<String>,
}

/// Key-value org configuration row from `org_settings` table.
#[derive(Debug, Clone, Serialize, sqlx::FromRow)]
pub struct OrgSetting {
    pub id: Uuid,
    pub org_id: Uuid,
    pub key: String,
    pub value: serde_json::Value,
    #[serde(with = "time::serde::rfc3339")]
    pub updated_at: OffsetDateTime,
}

#[derive(Debug, Deserialize, Validate)]
pub struct SetOrgSettingRequest {
    #[validate(length(max = 100))]
    pub key: String,
    pub value: serde_json::Value,
}

File: frontend/src/api/organization.ts
import { apiClient } from './client'

export interface Organization {
  id: string
  name: string
  slug: string
  timezone: string
  created_at: string
  updated_at: string
}

export interface OrgSetting {
  id: string
  org_id: string
  key: string
  value: unknown
  updated_at: string
}

export const organizationApi = {
  get: () =>
    apiClient.get<Organization>('/api/organization'),

  update: (body: { name?: string; timezone?: string }) =>
    apiClient.patch<Organization>('/api/organization', body),

  listSettings: () =>
    apiClient.get<OrgSetting[]>('/api/organization/settings'),

  setSetting: (body: { key: string; value: unknown }) =>
    apiClient.patch<OrgSetting>('/api/organization/settings', body),
}
